internal/pkg/provisioner/storageclass.go

```diff
@@ -99,29 +99,33 @@ func StorageClassReservedParams(sc *storagev1.StorageClass) map[string]string {
 }
 
 // ValidateOrSetStorageClassUID returns true if the StorageClass annotation on
-// the PVC object matches the uid passed in.
+// the PVC object matches the uid passed in or the StorageClass is younger than PVC.
 //
-// If the annotation does not exist, it sets the uid passed in as the new
-// StorageClass annotation.
-func ValidateOrSetStorageClassUID(ctx context.Context, k8s client.Client, uid types.UID, obj client.Object) (bool, error) {
-	provisionedUID := obj.GetAnnotations()[StorageClassUUIDAnnotationKey]
+// If the annotation does not exist and the StorageClass is younger than PVC,
+// it sets the uid passed in as the new StorageClass annotation.
+func ValidateOrSetStorageClassUID(ctx context.Context, k8s client.Client, sc client.Object, pvc client.Object) (bool, error) {
```

Comment:
nit: I'd keep the `client.Object` var name as `obj` instead of `pvc` as it makes it more clear what type it is.  Below where you use its key to fetch the PVC can it `pvc` instead of `v1Pvc`.

Reply:
We have two `client.Object`s at the moment, so somehow i wanted to inform clients which one is which.

Reply:
ah, I see. ok